src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/PackageSourceMapping.cs

```diff
@@ -38,7 +38,7 @@ public IReadOnlyList<string> GetConfiguredPackageSources(string packageId)
             return SearchTree.Value?.GetConfiguredPackageSources(packageId);
         }
 
-        internal PackageSourceMapping(Dictionary<string, IReadOnlyList<string>> patterns)
+        public PackageSourceMapping(Dictionary<string, IReadOnlyList<string>> patterns)
```

Comment:
Perhaps `IReadOnlyDictionary`?

Reply:
I think that wouldn't hurt given that the patterns are read only. 
The consideration here would be do we need a new object when the patterns are edited and I think the answer based on the usage in our code is yes. 

Curious what @kartheekp-ms and @erdembayar think.

Reply:
@[kartheekp-ms](https://github.com/kartheekp-ms) How can we move this along?

Reply:
@lbargaoanu - Sorry for the delay. I am okay to change the type to `IReadOnlyDictionary`. After this change the class may look like below. Please make this change and push a new commit so that I can run the CI.

``` cs
    public class PackageSourceMapping
    {
       /// Existing code

        /// <summary>
        /// Source name to package patterns list.
        /// </summary>
        internal IReadOnlyDictionary<string, IReadOnlyList<string>> Patterns { get; }

        private Lazy<SearchTree> SearchTree { get; }

       /// Existing code

        public PackageSourceMapping(IReadOnlyDictionary<string, IReadOnlyList<string>> patterns)
        {
            Patterns = patterns ?? throw new ArgumentNullException(nameof(patterns));
            IsEnabled = Patterns.Keys.Any();
            SearchTree = new Lazy<SearchTree>(() => GetSearchTree());
        }
```

Reply:
Unsure whether I caused the CI to break :)

Reply:
All the CI checks have passed. Approved this PR. Thank you for the contribution.